numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -1707,10 +1710,21 @@ mapiter_@name@(PyArrayMapIterObject *mit)
                 }
 
                 /*
-                 * Resetting is slow, so skip if the subspace iteration has
-                 * only a single inner loop.
+                 * Resetting is slow, so try to avoid resetting
+                 * if the subspace iteration is non-skipping.
+                 *
+                 * Resetting is avoided iif the next pointer points to
+                 * the correct location (identical to resetting).
+                 *
+                 * Also notice skip is kept as a `global` outside of the loop,
+                 * because the internal iteration of each external iterations,
+                 * share the same structure, if we are correct once we know
+                 * future iterations we are always correct.
+                 *
```

Comment:
No need for extra blank line. The comment is outdated with this version of the code.